### DISCUSSION TITLE
getAsset shows as deprecated, suggests getNFT, but that doesn't work and docs not updated

### DISCUSSION THREAD

**Discussion by rogerjbos** [ProjectOpenSea/opensea-js#1188] [General]
I was following the documentation and using `getAsset` as follows:
```
const asset = await openseaSDK.api.getAsset({
  tokenAddress: '0x4135063DC85190660ed08790f59Bc711d8b404c0', // string
  tokenId: null, // string | number | BigNumber | null
})
```
This function works fine, but according to VS Code it is deprecated, and suggests I should use `getNFT`.  I am wondering why the getting started docs have not been updated.  I tried to use the `getNFT` function, but I can't get it to work:
```
const asset = await openseaSDK.api.getNFT({chain: "ethereum", 
  address: "0x4135063DC85190660ed08790f59Bc711d8b404c0",
  identifier: "973"
})
```
Does anyone have any suggestions?  Does `getNFT` have a way to pass in null for identifier and get all the NFTs?

**Comment by thenerdassassin** [comment]
To get all the NFTs at a contract address or in collection use functions like getNFTsByCollection

https://github.com/ProjectOpenSea/opensea-js/blob/b4e640fa13dbf424cf89aa8bb1c986ad9cc957eb/src/api/api.ts#L395C33-L395C33

**Comment by ryanio** [comment]
> but I can't get it to work:

you are passing an object to getNFT instead of passing the individual parameters. as @thenerdassassin said, you can use getNFTsByCollection and getNFTsByContract.

we are in the process of updating our docs and will have a new release for them soon, thanks for your patience.

**Reply by rogerjbos** [reply]
Thank you both for the responses.  Ah yes, it was the object that was the problem.  Thank you so much.  This works:

```
const asset = await openseaSDK.api.getNFTsByContract(Chain.Mainnet, 
  "0x4135063DC85190660ed08790f59Bc711d8b404c0")
```
